code/modules/client/preferences/entries/character/quirkoptions.dm

```diff
@@ -1,13 +1,28 @@
-/datum/preference/choiced/quirk_prosthetic_limb_location
+/datum/preference/choiced/quirk
 	category = PREFERENCE_CATEGORY_NON_CONTEXTUAL
 	preference_type = PREFERENCE_CHARACTER
+	can_randomize = TRUE
+	abstract_type = /datum/preference/choiced/quirk
+	var/required_quirk_name // name of the quirk to be checked for (caps sensitive)
```

Comment:
if you use `///` triple slash comments on the line before it has better VSCode support as hovering the var name will show the comment